### ISSUE TITLE
Improve window auto scrolling behavior

### ISSUE THREAD

**Issue by mdiehl216** [wxMaxima-developers/wxmaxima#1947]
When an input cell is just at the bottom of the window, the standard behavior after hitting enter is that the window scrolls up automatically. But it scrolls up so far that the previous input and its output are out of vision.

This often undesirable. After hitting enter, I usually want to see the output.

I use Windows OS.

Desirable is that the window scrolls up so far that either the last input cell or at least its output are visible at the top of the window.

Thank you!

**Comment by robert-dodier** [comment]
Thank you for the bug report. Looks like this problem is related to #1939.

**Comment by mdiehl216** [comment]
Yes. It seems to me that in both bug reports the scrolling is indicated correctly, only the final scrolling position is wrong.

**Comment by gunterkoenigsmann** [comment]
The problem is that I never found out how scrolling actually works...

**Comment by robert-dodier** [comment]
`git grep scroll` shows that scrolling is mentioned in various places in the code base; the problem is just to identify the place or places where scrolling is being carried out in order to expose stuff at the bottom of the window, and make the scroll distance smaller (e.g. scroll only enough to make the new cell visible). I will try to look at that in the near future although if someone else can cover it, that would be terrific.

**Comment by mdiehl216** [comment]
Since the related bug mentions the scrolling problem happening at long
output, it may be that there is a test that measures if the current output
fits into the current window. If not, the window is scrolled.
So the problem is not necessarily that the cell is at the bottom of the
window. In both cases (long output, cell at bottom of the window) scrolling
is necessary. It's just that the final position is wrong.

**Comment by gunterkoenigsmann** [comment]
This issue should be resolved now.
